crawler/main.py

```diff
@@ -0,0 +1,44 @@
+import os
+from crawler import Crawler
+from database import Database
+from concurrent.futures import ThreadPoolExecutor
+
+
+class Worker:
+
+    def __init__(self, crawler: Crawler, database: Database, module_id: str):
+        self.crawler = crawler
+        self.database = database
+        self.module_id = module_id
+
+    def run(self):
+        for mapping in self.crawler.module_degree_mappings(self.module_id, english=True):
+            # TODO: add mapping to database
+            pass
+
+
+def run_crawler(self, max_workers=1):
```

Comment:
What is self here? The function is not contained in a class?

Reply:
> What is self here? The function is not contained in a class?

💀